end_to_end_tests/golden-record-custom/custom_e2e/models/a_model.py

```diff
@@ -155,35 +234,125 @@ def _parse_a_camel_date_time(data: Any) -> Union[datetime.datetime, datetime.dat
+                if not isinstance(_not_required_one_of_models, Unset):
+                    not_required_one_of_models = FreeFormModel.from_dict(_not_required_one_of_models)
+
+                return not_required_one_of_models
+            except:  # noqa: E722
+                pass
+            if not isinstance(data, dict):
+                raise TypeError()
+            not_required_one_of_models = UNSET
+            _not_required_one_of_models = data
+            if not isinstance(_not_required_one_of_models, Unset):
+                not_required_one_of_models = ModelWithUnionProperty.from_dict(_not_required_one_of_models)
+
+            return not_required_one_of_models
+
+        not_required_one_of_models = _parse_not_required_one_of_models(d.pop("not_required_one_of_models", UNSET))
+
+        def _parse_not_required_nullable_one_of_models(
+            data: Union[Unset, None, Dict[str, Any], str]
+        ) -> Union[Unset, None, FreeFormModel, ModelWithUnionProperty, str]:
+            not_required_nullable_one_of_models: Union[Unset, None, FreeFormModel, ModelWithUnionProperty, str]
+            if data is None:
+                return data
+            if isinstance(data, Unset):
+                return data
+            try:
+                if not isinstance(data, dict):
+                    raise TypeError()
+                not_required_nullable_one_of_models = UNSET
+                _not_required_nullable_one_of_models = data
+                if not isinstance(_not_required_nullable_one_of_models, Unset):
+                    not_required_nullable_one_of_models = FreeFormModel.from_dict(_not_required_nullable_one_of_models)
+
+                return not_required_nullable_one_of_models
+            except:  # noqa: E722
+                pass
+            try:
+                if not isinstance(data, dict):
+                    raise TypeError()
+                not_required_nullable_one_of_models = UNSET
+                _not_required_nullable_one_of_models = data
+                if not isinstance(_not_required_nullable_one_of_models, Unset):
+                    not_required_nullable_one_of_models = ModelWithUnionProperty.from_dict(
+                        _not_required_nullable_one_of_models
+                    )
+
+                return not_required_nullable_one_of_models
+            except:  # noqa: E722
+                pass
+            return cast(Union[Unset, None, FreeFormModel, ModelWithUnionProperty, str], data)
+
+        not_required_nullable_one_of_models = _parse_not_required_nullable_one_of_models(
+            d.pop("not_required_nullable_one_of_models", UNSET)
+        )
+
         nullable_model = None
         _nullable_model = d.pop("nullable_model")
         if _nullable_model is not None:
-            nullable_model = AModelNullableModel.from_dict(cast(Dict[str, Any], _nullable_model))
+            nullable_model = AModelNullableModel.from_dict(_nullable_model)
```

Comment:
Yeah, it's because `Any` has weird semantics, in that it basically just means "ignore any type errors". If we were to change the signature to `object`, then it would error.